web-api/src/dispatchers/ses/sendEmailWithAttachment.ts

```diff
@@ -0,0 +1,89 @@
+import { SendRawEmailCommand, SESClient } from '@aws-sdk/client-ses';
+import { type ServerApplicationContext } from '@web-api/applicationContext';
+import { basename } from 'path';
+import { existsSync, readFileSync } from 'fs';
+
+export const sendEmailWithAttachment = async ({
+  applicationContext,
+  body,
+  filePath,
+  recipient,
+  subject,
+}: {
+  applicationContext: ServerApplicationContext;
+  body: string;
+  filePath: string;
+  recipient: string;
+  subject: string;
+}): Promise<void> => {
+  if (!body || !filePath || !recipient || !subject) {
+    throw new Error(
+      'Error sending email: missing recipient, body, subject, or attachment.',
+    );
+  }
+  if (!existsSync(filePath)) {
+    throw new Error('Error sending email: attachment not found.');
+  }
+  const sender = applicationContext.environment.emailFromAddress;
+  const sesClient: SESClient = applicationContext.getEmailClient();
+  const fileContent = readFileSync(filePath);
+
+  const params = {
+    RawMessage: {
+      Data: buildRawEmail({
+        body,
+        fileContent,
+        filename: basename(filePath),
+        recipient,
+        sender,
+        subject,
+      }),
+    },
+  };
+  const rawEmailCommand = new SendRawEmailCommand(params);
+
+  const result = await sesClient.send(rawEmailCommand);
+  console.log('Email sent:', result);
+};
+
+const buildRawEmail = ({
+  body,
+  fileContent,
+  filename,
+  recipient,
+  sender,
+  subject,
+}: {
+  body: string;
+  fileContent: Buffer;
+  filename: string;
+  recipient: string;
+  sender: string;
+  subject: string;
+}): Uint8Array => {
+  const boundary = '----=_Part_0_123456789.123456789';
+  const rawEmail = [
+    `From: ${sender}`,
+    `To: ${recipient}`,
+    `Subject: ${subject}`,
```

Comment:
We might need to add some additional headers here - replyTo/bounce handler, sender verification headers, etc.

Reply:
Update: checked into this with @mmarcotte, and those are set at the SES identity level, not on individual messages. Woohoo!